### ISSUE TITLE
[Suggestion] Automate checking versions without extra work on other modders

### ISSUE THREAD

**Issue by TelepathicGrunt** [JamCoreModding/Notify#3]
So it has begun where people are now poking me to add support for Notify lol
![image](https://user-images.githubusercontent.com/40846040/142013017-bad49417-8dae-4b57-9486-f59dff09f7bc.png)

However, from my understanding of this mod, I have to make a json file and manually edit it every time I update to make notify work. The issue is, I will forget to do this extra step every single time I update lol. It's not going to work out well for me.

So I got two suggestion that could work. One being to use the URL listed in the homepage section of the fabric.mod.json which the majority of mods have and then check that curseforge or modrinth page for if a newer version was posted than the current version on. This would sue the modrinth/curseforge api of course from the given URL. Example: https://github.com/TelepathicGrunt/Blame/blob/962d40a0d692c1716ccaef1c3368b66968dcf489/src/main/resources/fabric.mod.json#L12 

A second idea that could be done is the custom json is formatted like this instead:
```
"custom": {
    "notify_version_url": "https://github.com/TelepathicGrunt/Blame/blob/Fabric-1.17.x/gradle.properties"
    "notify_version_key": "mod_version"
}
```
Then it parses the gradle.properties file linked for the key and read it's version to see if it is newer. Of course, that second way only works if the modder changes the mod_version only when they release an update. 

Basically, I would be totally fine doing compat with your mod as long as it is a do-once-and-forget kind of thing instead of having an extra file I have to edit every update. Or did I misunderstand how this mod works?

**Comment by Jamalam360** [comment]
Yes, I was planning on adding support for the MR/CF APIs at some point. I also like the gradle.properties idea. so perhaps I could support both. I was aware when I was making it that this would be an issue, but I hope that if it gets fixed it should be a viable mod

**Comment by Jamalam360** [comment]
I've done the inital work for this, it's still a bit iffy reliability wise with Modrinth because I'm just getting the absolute latest version, not specifying Minecraft, but I'll fix that. CurseForge isn't supported yet because....I didn't want to deal with that yet. I would say the gradle.properties method is the best at the moment
